### ISSUE TITLE
Solid objects incorrectly rendered with transparency when ReShade enabled.

### ISSUE THREAD

**Issue by Nerrel** [gonetz/GLideN64#1802]
-Windows 10x64, Project64, GTX 970

This glitch occurs using the WIP build "GLideN64_6a10a83d" posted below, but not in the 3.0 release or the Naboo Alpha 1 release.  In the milk bar in Majora's Mask the bar, stools, Link, and other solid objects are transparent.  Textures along the walls are also displayed incorrectly:
![Image of transparent bar](https://c1.staticflickr.com/1/811/26722221807_f827c9335f_c.jpg)

Some objects still render correctly, but changing the camera position can cause them to glitch.  The far back wall is correctly displayed in the first image, but moving the camera up triggers the problem:
![image of correct wall](https://c1.staticflickr.com/1/786/41550049962_3dce292691_c.jpg)
![image of broken wall](https://c1.staticflickr.com/1/934/26722218847_0103a4963f_c.jpg)  

I have not tested extensively with this build, but Link seems to turn transparent in almost every indoor environment in clock town (he remains solid outdoors).  His other masks/forms do not turn transparent, although the Goron's necklace alone _does_. Environmental textures and objects also break in areas outside of the milk bar, but not as frequently (the fairy fountain and item shop have the issue clearly).  

Here is a savestate from within the bar:
https://mega.nz/#!vV9yiSzZ!1fYVkMPjnWmIE51Mnk7-1wu7YJnMc8PTmWNvfEMOZHE
And the build the issue was found in:
https://drive.google.com/file/d/1YjBzzUZ7dT3WVh7a4VEf-ozJXY6j0Xz-/view

**Comment by loganmc10** [comment]
A couple questions:

Do you have N64 Depth Compare enabled? What GPU and OS do you have?

**Comment by Nerrel** [comment]
Depth compare is not enabled.  As far as I'm aware, every setting is on the default/recommended option aside from the resolution and AA/AF.  Using custom textures doesn't make a difference.  I'm using Windows 10x64 and a GTX 970 on Project64, sorry I didn't add that to the first post.  I don't have Mupen64 installed right now to test with.

It seems like the textures that are most commonly impacted are the grayscale alpha textures.  In a lot of areas, regular RGB textures display correctly but the alpha textures that overlay objects don't.  This doesn't seem true across the board, though.

**Comment by loganmc10** [comment]
Hm ok, I tested on my Linux laptop (mupen64plus) with your save state and didn't see that issue, I'll do some more testing.

Are your GPU drivers up to date?

**Comment by Nerrel** [comment]
Yep, I'm using the newest 391.35 driver.

**Comment by loganmc10** [comment]
Just tested my phone as well, can't see any issues

**Comment by gonetz** [comment]
@Nerrel do you have that issue with texture packs and enhancements disabled?

**Comment by Nerrel** [comment]
I removed the config file from my plugins folder to force GlideN64 to completely default on every setting and I made sure ReShade was disabled.  There was no change:
![image of fairy fountain](https://c1.staticflickr.com/1/792/39785021110_234d5722e6_c.jpg)

**Comment by fzurita** [comment]
It looks normal in Android with an Adreno 540 GPU. I wonder which commit broke it, it must had been recent.

I'll check my Shield TV which also has a NVidia GPU.

**Comment by fzurita** [comment]
No problem in the Shield TV either.

**Comment by oddMLan** [comment]
Try disabling the Shader Cache

**Comment by gonetz** [comment]
> I made sure ReShade was disabled

It is enabled on your screen shot.

**Comment by Nerrel** [comment]
^That turned out to be the problem, I thought disabling ReShade from its interface would eliminate it as a factor but clearly it didn't.  After removing the ReShade installation completely the game worked normally again.  I tried rolling back to a different version of ReShade but got the same glitches again, even with no effects installed.  Whatever was changed between the Naboo Alpha 1 release and now seems to have broken ReShade compatibility...

It's not really a GlideN64 specific problem then, but I'd still appreciate it if it compatibility could be restored at some point.  Can anyone install the latest ReShade and confirm that it breaks the rooms posted above in the same way as my system?

**Comment by gonetz** [comment]
Are you sure that ReShade is completely removed? Check that opengl32.dll does not present in your emulator folder. Nobody else can reproduce that bug.

**Comment by Nerrel** [comment]
The glitch doesn't happen when ReShade's opengl.dll file is removed; the plugin works correctly when used by itself without ReShade installed.  In that sense, this turned out not to be a problem with GlideN64 itself.  

It's just that ReShade compatibility is broken for those who want to use that program.  You would have to install ReShade (it doesn't seem to matter which version) to reproduce the problem.   Again, I'm not sure if you consider fixing ReShade compatibility a priority, but whatever change broke it was made recently.

**Comment by gonetz** [comment]
I would like to maintain compatibility with ReShade, but I have no resources for that.
Thus, no ReShade unless somebody else will take that task.

**Comment by Nerrel** [comment]
I wanted to update this by saying that games can't run at all now when ReShade is installed.  GlideN64 gives an error message reading "unable to find a suitable pixel format" at startup and then Project64 gives this error: 
![image of error](https://c2.staticflickr.com/8/7855/33325451068_deddb60f15_o.png)

No log is generated, so I don't have any more information than that.  It doesn't seem limited to GlideN64 though,  Project64 Video also crashes with an "error setting display mode" message followed by this error: 
![image of pj error](https://c2.staticflickr.com/8/7871/33262892388_8b8ca9ab37_o.jpg)

Pj64 Video does generate a log referencing an invalid memory address over and over.  
[Project64.log](https://github.com/gonetz/GLideN64/files/2898406/Project64.log)

Again, this isn't a priority and it's not clear whether the problem lies with with GlideN64 or Project64 or ReShade, but it's been pretty hard to find any information about the cause of this issue.  Aside from the transparency glitches, ReShade was working with the WIP GlideN64 build I listed when I opened this issue and games could still load, but at some point the more recent WIP builds broke any kind of compatibility at all.  I just wanted to ask if anything stands out as a problem when considering this...
